pkg/controller/vpc_nat_gateway.go

```diff
@@ -0,0 +1,818 @@
+		util.AttachmentNetworkAnnotation: fmt.Sprintf("%s/%s", c.config.PodNamespace, util.VpcExternalNet),
+		util.LogicalSwitchAnnotation:     gw.Spec.Subnet,
+		util.IpAddressAnnotation:         gw.Spec.LanIp,
+	}
+
+	ovnControllerPod, err := c.config.KubeClient.CoreV1().Pods(c.config.PodNamespace).
+		Get(context.Background(), c.config.PodName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("failed to get kube-ovn-controller deployment.")
+		return nil, err
+	}
+	image := ovnControllerPod.Spec.Containers[0].Image
+
+	dp = &v1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      labels,
+					Annotations: podAnnotations,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "vpc-gat-gw",
+							Image:           image,
+							Command:         []string{"bash"},
+							Args:            []string{"-c", "while true; do sleep 10000; done"},
+							ImagePullPolicy: corev1.PullAlways,
+							SecurityContext: &corev1.SecurityContext{
+								Privileged:               &privileged,
+								AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+							},
+						},
+					},
+				},
+			},
+			Strategy: v1.DeploymentStrategy{
+				Type: v1.RecreateDeploymentStrategyType,
+			},
+		},
+	}
+	return
+}
+
+func (c *Controller) cleanUpVpcNatGw() error {
+	gws, err := c.vpcNatGatewayLister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("failed to get vpc nat gateway, %v", err)
+		return err
+	}
+	for _, gw := range gws {
+		c.delVpcNatGatewayQueue.Add(gw.Name)
+	}
+	return nil
+}
+
+func (c *Controller) getNatGwPod(name string) (*corev1.Pod, error) {
+	sel, _ := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: map[string]string{"app": genNatGwDpName(name)},
+	})
+
+	pods, err := c.podsLister.Pods(c.config.PodNamespace).List(sel)
+	if err != nil {
+		return nil, err
+	} else if len(pods) != 1 {
+		return nil, fmt.Errorf("too many pod.")
+	} else if pods[0].Status.Phase != "Running" {
+		return nil, fmt.Errorf("pod is not active now")
+	}
+
+	return pods[0], nil
+}
+
+func (c *Controller) gcVpcExternalNetwork() (err error) {
+	networkClient := c.config.AttachNetClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions(c.config.PodNamespace)
+	_, err = networkClient.Get(context.Background(), util.VpcExternalNet, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		if k8serrors.IsForbidden(err) {
+			klog.Warningf("failed to get net-attach-def, %v", err)
+			return nil
+		}
+		return err
+	}
+	err = networkClient.Delete(context.Background(), util.VpcExternalNet, metav1.DeleteOptions{})
+	if k8serrors.IsForbidden(err) {
+		klog.Warningf("failed to delete net-attach-def, %v", err)
+		return nil
```

Comment:
If Multus-CNI is not deployed, the error log may be printed all the time. I don't think it will matter if it is not cleaned up here, so only the warning log is printed.